tools/cndump/cndb_dump.c

```diff
@@ -0,0 +1,84 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "cndb_reader.h"
+#include "cndb_record.h"
+#include "globals.h"
+#include "fatal.h"
+
+/* command line options for cndb sub-command */
+struct opt {
+    bool oneline;               // print each record on a single line
+};
+
+static struct opt opt;
+
+static void
+help(const char *progname)
+{
+    printf("Usage: %s cndb [options] <kvdb_home>\n", progname);
+    printf("Pretty print a KVDB's CNDB log.\n"
+        "Options:\n"
+        "  -h    print help\n"
+        "  -l    print each CDNB record on a single line\n");
+}
+
+unsigned int
+cndb_cmd_parse_args(int argc, char **argv)
+{
+    unsigned int consumed;
+    int c;
+
+    /* argv[0] should hold command name */
+    assert(argc > 0);
+    consumed = 1;
+    optind = 0;
+
+    while ((c = getopt(argc, argv, "+:hl")) != -1) {
+        consumed++;
+        switch (c) {
+        case 'h':
+            help(argv[0]);
+            exit(0);
+        case 'l':
+            opt.oneline = true;
+            break;
+        default:
+            syntax("invalid option -%c", c);
```

Comment:
I prefer this concise output:
```
invalid option: -X, use -h for help
```

Reply:
Why tell a user to print help, instead of just printing the help? Either way this is just another source of inconsistency for us.

Reply:
Because most of the time I can look at the one line error and know how to fix it.  I really really really hate it when a program blast ten pages of usage info at me that I didn't ask for, and then I have to wade through it to look for an error message telling me what was wrong, only to find it didn't tell me anything whatsoever about what was wrong, but so very helpfully scrolled off the important stuff on my screen for a bunch of info I didn't ask for...   Don't get me going on this!!!  :-)

Reply:
BTW, you really should handle ':' and '?".  If someone adds an option that takes an arg then your default case will give incorrect and unhelpful information in some error cases.  See kmt or ksync for examples...

Reply:
Using `optopt`, adding case for `:`, but leaving `?` with default to print "invalid option" message.  Don't get why you would say "option ignored" as you do with kmt.

Reply:
getopt returns '?' for any given option not present in the opt string.  But if say I add a 'q' option to the optstring, but don't supply a case for it, then I wind up in the default case.  I could fail or whatever, but it's just a harmless oversight on the programmer's end.  Point being, 'q' is not an invalid option insofar as getopt is concerned, it's a logic error, so fail if you wanna...